samza-core/src/main/java/org/apache/samza/clustermanager/ClusterBasedJobCoordinator.java

```diff
@@ -172,19 +235,25 @@ public void run() {
     }
   }
 
+  private boolean hasCallbackException() throws Exception {
+    if (coordinatorException != null) {
+      throw coordinatorException;
```

Comment:
Make sense. Will change it to return false to stop the loop.

Reply:
Actually, I will keep the original function but change name to checkAndThrowException(). I also put the check before the check on isInterrupted as below:
while (!containerProcessManager.shouldShutdown() && !checkAndThrowException() && !isInterrupted)

s.t. if there is an coordinatorException, it will be thrown and logged even the thread is interrupted.